features/map.go

```diff
@@ -80,18 +81,23 @@ func createMapTypeAttr(mt ebpf.MapType) *sys.MapCreateAttr {
 		btfKeyTypeID = 1   // BTF_KIND_INT
 		btfValueTypeID = 3 // BTF_KIND_ARRAY
 		btfFd = ^uint32(0)
+	case ebpf.StructOpsMap:
+		// StructOps requires setting a vmlinux type id, but id 1 will always
+		// be some type of integer. This will cause ENOTSUPP.
+		btfVmlinuxValueTypeID = 1
```

Comment:
How does setting this not cause EBADF / EINVAL from btf_get_by_fd? https://elixir.bootlin.com/linux/v5.18.14/source/kernel/bpf/syscall.c#L890

Reply:
[`find_map_and_alloc()`](https://elixir.bootlin.com/linux/v5.18.14/source/kernel/bpf/syscall.c#L864) will return the `ENOTSUPP` when calling [`map_alloc()`](https://elixir.bootlin.com/linux/v5.18.14/source/kernel/bpf/syscall.c#L129) I think.